paimon-core/src/main/java/org/apache/paimon/table/system/SchemasTable.java

```diff
@@ -221,26 +222,28 @@ public InnerTableRead withFilter(Predicate predicate) {
             if (predicate instanceof CompoundPredicate) {
                 CompoundPredicate compoundPredicate = (CompoundPredicate) predicate;
                 if ((compoundPredicate.function()) instanceof And) {
-                    List<Predicate> children = compoundPredicate.children();
-                    for (Predicate leaf : children) {
-                        handleLeafPredicate(leaf, leafName);
-                    }
+                    PredicateUtils.traverseCompoundPredicate(
+                            predicate,
+                            leafName,
+                            (Predicate p) -> {
+                                handleLeafPredicate(p, leafName);
+                            },
```

Comment:
Can `handleLeafPredicate` method also be extracted to the util?

Reply:
I had considered about it，this method common logic is less，extract may cost more codes，so keep as current maybe better. Thanks~ @LinMingQiang